smack-experimental/src/main/java/org/jivesoftware/smackx/chat_markers/ChatMarkersManager.java

```diff
@@ -218,4 +225,45 @@ public synchronized boolean removeIncomingChatMarkerMessageListener(ChatMarkersL
         }
         return res;
     }
+
+    /**
+     * Send a message stanza to the recipient defined in the <tt>To</tt> getter from the <tt>Message</tt>.
+     *
+     * @param message          instance of {@link Message} with a To previous defined.
+     * @param chatMarkersState one of the values given in {@link ChatMarkersState}.
+     * @param messageId        id of the message to be updated.
+     * @throws NotConnectedException if the connection is not connected.
+     * @throws InterruptedException  if the connection is interrupted.
+     * @throws IllegalStateException if one of the params don't match the rules.
+     */
+    public void markMessage(Message message, ChatMarkersState chatMarkersState, String messageId)
```

Comment:
I would expect that the user gives as a message with a markable extension to such a method. Then you could remove the `ChatMarkersState` and `String` (`messageId`) parameters, which would make the method much more user friendly.

Reply:
but the user would need to deal with `message.addExtension(new ChatMarkersElements.ReceivedExtension(messageId));` by himself, in other words he would doing some of the job that this class has to do.

What I want is class that can be used almost out of the box but of course knowing how the XEP works.